code/game/objects/items/stacks/sheets/sheet_types.dm

```diff
@@ -57,6 +57,20 @@ var/global/list/datum/stack_recipe/metal_recipes = list ( \
 	flags = CONDUCT
 	origin_tech = "materials=1"
 
+/obj/item/stack/sheet/metal/attackby(obj/item/weapon/W, mob/living/user)
+	..()
+	if(istype(W, /obj/item/weapon/paper/talisman/construction))
+		if(amount>24)
```

Comment:
Sounds like you didn't typecast the metal properly; if you're doing the below stuff, as an example, you'd have to do `if(target.use(25))`, where the target is typecasted(as in, byond considers it to be something) to be metal sheets.